sdk/storage/storage-queue/karma.conf.js

```diff
@@ -84,22 +84,31 @@ module.exports = function(config) {
 
     jsonToFileReporter: {
       filter: function(obj) {
-        if (obj.writeFile) {
-          const fs = require("fs-extra");
-          // Create the directories recursively incase they don't exist
-          try {
-            // Stripping away the filename from the file path and retaining the directory structure
-            fs.ensureDirSync(obj.path.substring(0, obj.path.lastIndexOf("/") + 1));
-          } catch (err) {
-            if (err.code !== "EEXIST") throw err;
-          }
-          fs.writeFile(obj.path, JSON.stringify(obj.content, null, " "), (err) => {
-            if (err) {
-              throw err;
+        // - jsonToFileReporter filters the JSON strings in console.logs.
+        // - Console logs with `.writeFile` property are captured and are written to a file(recordings).
+        // - The other console statements are captured and printed normally.
+        // - Example - console.warn("hello"); -> console.log({ warn: "hello" });
+        // - Example - console.log("hello"); -> console.log({ log: "hello" });
+        if (process.env.TEST_MODE === "record") {
+          if (obj.writeFile) {
+            const fs = require("fs-extra");
+            // Create the directories recursively incase they don't exist
+            try {
+              // Stripping away the filename from the file path and retaining the directory structure
+              fs.ensureDirSync(obj.path.substring(0, obj.path.lastIndexOf("/") + 1));
+            } catch (err) {
+              if (err.code !== "EEXIST") throw err;
             }
-          });
+            fs.writeFile(obj.path, JSON.stringify(obj.content, null, " "), (err) => {
+              if (err) {
+                throw err;
+              }
+            });
+          } else {
+            console.log(obj);
```

Comment:
A simple suggestion (for a future PR maybe?):

We are passing:
```
{
  method_name: some_stuff
}
```
If we wanted to keep the logging behavior identical to the original one, we could do:
```
{
  method: method_name,
  content: some_stuff
}
```
And call `console[obj.method](obj.content)` here. Maybe add a check to avoid unexpected exceptions.

Reply:
Future PR  - https://github.com/Azure/azure-sdk-for-js/issues/4205